evm/test/RateLimit.t.sol

```diff
@@ -639,7 +641,189 @@ contract TestRateLimit is Test, IRateLimiterEvents {
+            assertEq(
+                outboundLimitParams.currentCapacity.getAmount(),
+                outboundLimitParams.limit.sub(transferAmount).getAmount()
+            );
+            assertEq(outboundLimitParams.lastTxTimestamp, sendAgainTime);
+        }
+
+        {
+            // assert that the inbound limit is at max again (because of backflow)
+            IRateLimiter.RateLimitParams memory inboundLimitParams =
+                nttManager.getInboundLimitParams(TransceiverHelpersLib.SENDING_CHAIN_ID);
+            assertEq(
+                inboundLimitParams.currentCapacity.getAmount(), inboundLimitParams.limit.getAmount()
+            );
+            assertEq(inboundLimitParams.lastTxTimestamp, sendAgainTime);
+        }
+    }
+
+    // helper functions
+    function setupToken() public returns (address, address, DummyToken, uint8) {
+        address user_A = address(0x123);
+        address user_B = address(0x456);
+
+        DummyToken token = DummyToken(nttManager.token());
+
+        uint8 decimals = token.decimals();
+        assertEq(decimals, 18);
+
+        return (user_A, user_B, token, decimals);
+    }
+
+    function initializeTransceivers() public returns (ITransceiverReceiver[] memory) {
+        (DummyTransceiver e1, DummyTransceiver e2) =
+            TransceiverHelpersLib.setup_transceivers(nttManager);
+
+        ITransceiverReceiver[] memory transceivers = new ITransceiverReceiver[](2);
+        transceivers[0] = e1;
+        transceivers[1] = e2;
+
+        return transceivers;
+    }
+
+    // transfer tokens from user_A to user_B
+    // this consumes capacity on the outbound side
+    // send tokens from user_B to user_A
+    // this consumes capacity on the inbound side
+    // send tokens from user_A to user_B
+    // this should consume capacity on the outbound side
+    // and backfill the inbound side
+    function testFuzz_CircularFlowBackFilling(uint64 mintAmt, uint256 transferAmt) public {
+        mintAmt = uint64(bound(mintAmt, 2, type(uint256).max));
+        transferAmt = uint64(bound(transferAmt, 1, mintAmt - 1));
+
+        (address user_A, address user_B, DummyToken token, uint8 decimals) = setupToken();
+
+        TrimmedAmount memory mintAmount = TrimmedAmount(mintAmt, 8);
+        token.mintDummy(address(user_A), mintAmount.untrim(decimals));
+        nttManager.setOutboundLimit(mintAmount.untrim(decimals));
+
+        // transfer 10 tokens
+        vm.startPrank(user_A);
+
+        // TrimmedAmount memory transferAmount = TrimmedAmount(transferAmt, 8);
+        token.approve(address(nttManager), type(uint256).max);
+
+        // TODO: also fuzz the fromDecimals?
+
+        // allow for amounts greater than uint64 to check if [`transfer`] reverts
+        // on amounts greater than u64 MAX.
+        TrimmedAmount memory transferAmount = transferAmt.trim(decimals, 8);
+
+        // check error conditions
+        if (transferAmount.amount == 0) {
+            vm.expectRevert();
```

Comment:
Why don't we use the actual revert signature?

Reply:
Also btw @RahulMaganti47 I prefer using `abiEncodeWithSelector` rather than `abiEncodeWithSignature` as then you don't have to copy the error signature string around. Calling that out as I noticed in the `setIsWormholeEvmChain` PR

Reply:
ah nice, yea thats cleaner. 

yea we should use the revert signature (it hasn't a standard that we've enforced). there are a few places, esp. in the integration tests where the revert signatures are omitted. i can address those in a subsequent PR